lib/relationships/relationships.ts

```diff
@@ -0,0 +1,279 @@
+			},
+			to: {
+				type: '*',
+			},
+		},
+	},
+];
+
+// From jellyfish-plugin-channels
+// view@1.0.0 is attached to channel@1.0.0
+const relationshipsFromPluginChannels = [
+	{
+		slug: `relationship-view-is-attached-to-channel`,
+		type: 'relationship@1.0.0',
+		name: 'is attached to',
+		data: {
+			inverseName: 'has attached element',
+			title: 'View',
+			inverseTitle: 'Channel',
+			from: {
+				type: 'view@1.0.0',
+			},
+			to: {
+				type: 'channel@1.0.0',
+			},
+		},
+	},
+];
+
+// Found while running e2e:ui
+const relationshipsFromFailedE2e = [
+	{
+		slug: `relationship-contact-is-attached-to-user`,
+		type: 'relationship@1.0.0',
+		name: 'is attached to user',
+		data: {
+			// The one defined on the SDK has name "has attached contact" instead of "has contact"
+			inverseName: 'has contact',
+			title: 'Contact',
+			inverseTitle: 'User',
+			from: {
+				type: 'contact@1.0.0',
+			},
+			to: {
+				type: 'user@1.0.0',
+			},
+		},
+	},
+	{
+		slug: `relationship-message-is-attached-to-thread`,
+		type: 'relationship@1.0.0',
+		name: 'is attached to',
+		data: {
+			inverseName: 'has attached element',
+			title: 'Message',
+			inverseTitle: 'Thread',
+			from: {
+				type: 'message@1.0.0',
+			},
+			to: {
+				type: 'thread@1.0.0',
+			},
+		},
+	},
+	{
+		slug: `relationship-message-is-attached-to-issue`,
+		type: 'relationship@1.0.0',
+		name: 'is attached to',
+		data: {
+			inverseName: 'has attached element',
+			title: 'Message',
+			inverseTitle: 'Issue',
+			from: {
+				type: 'message@1.0.0',
+			},
+			to: {
+				type: 'issue@1.0.0',
+			},
+		},
+	},
+	{
+		slug: `relationship-message-is-attached-to-support-thread`,
+		type: 'relationship@1.0.0',
+		name: 'is attached to',
+		data: {
+			inverseName: 'has attached element',
+			title: 'Message',
+			inverseTitle: 'Support Thread',
+			from: {
+				type: 'message@1.0.0',
+			},
+			to: {
+				type: 'support-thread@1.0.0',
+			},
+		},
+	},
+	{
+		slug: `relationship-whisper-is-attached-to-support-thread`,
+		type: 'relationship@1.0.0',
+		name: 'is attached to',
+		data: {
+			inverseName: 'has attached element',
+			title: 'Whisper',
+			inverseTitle: 'Support Thread',
+			from: {
+				type: 'whisper@1.0.0',
+			},
+			to: {
+				type: 'support-thread@1.0.0',
+			},
+		},
+	},
+	{
+		slug: `relationship-whisper-is-attached-to-card`,
+		type: 'relationship@1.0.0',
+		name: 'is attached to',
+		data: {
+			inverseName: 'has attached element',
+			title: 'Whisper',
+			inverseTitle: 'Card',
+			from: {
+				type: 'whisper@1.0.0',
+			},
+			to: {
+				type: 'card@1.0.0',
+			},
+		},
+	},
+];
+
+/**
+ * Creates an array of relationships from sdk link constraints.
+ *
+ * SDK constraints come in pairs, with the "data.reverse" property being equal to the slug of the inverse
+ * Relationships are bidirectional and thus contain info from two sdk-link-constraints
+ * @param constraints
+ * @returns Relationship[]
+ */
+const convertSdkLinkContraintsToRelationships = (
```

Comment:
I don't think we should reference the sdk here, its farther up the stack and the worker technically shouldn't know it exists.